drivers/pinctrl/pinctrl_gd32_af.c

```diff
@@ -16,9 +17,14 @@ BUILD_ASSERT((GD32_OTYPE_PP == GPIO_OTYPE_PP) &&
 	     "pinctrl output type definitions != HAL definitions");
 
 BUILD_ASSERT((GD32_OSPEED_2MHZ == GPIO_OSPEED_2MHZ) &&
+#ifdef CONFIG_SOC_SERIES_GD32F3X0
+	     (GD32_OSPEED_10MHZ == GPIO_OSPEED_10MHZ) &&
+	     (GD32_OSPEED_50MHZ == GPIO_OSPEED_50MHZ),
+#else
 	     (GD32_OSPEED_25MHZ == GPIO_OSPEED_25MHZ) &&
 	     (GD32_OSPEED_50MHZ == GPIO_OSPEED_50MHZ) &&
 	     (GD32_OSPEED_200MHZ == GPIO_OSPEED_200MHZ),
+#endif /* CONFIG_SOC_SERIES_GD32F3X0 */
```

Comment:
See comments about HAS_FOO

Reply:
Sorry, cloud you please give the path? I not sure what's happen in here.

Reply:
It looks like there may be multiple GPIO IP versions on GD devices (that should explain why different speeds are supported and why they use different register values). It would be nice to clarify that with vendor or build a table for all series to see the _pattern_.

Reply:
> Sorry, cloud you please give the path? I not sure what's happen in here.

It is same principle mentioned here https://github.com/zephyrproject-rtos/zephyr/pull/40283#discussion_r759699799

Reply:
```c
#ifndef CONFIG_SOC_SERIES_GD32F3X0
        (GD32_OSPEED_200MHZ == GPIO_OSPEED_200MHZ) &&
#endif /* !CONFIG_SOC_SERIES_GD32F3X0 */
        1U,
```

Reply:
@cameled please check previous comment (avoids treating 10/50 differently from 2).

Reply:
Hi @gmarull thanks for your advice, please check again, hope I get the point.